src/http/parse_url.cpp

```diff
@@ -55,5 +55,23 @@ ErrorOr<Url> parse_url_noexcept(std::string url) {
     }
 }
 
+std::string Url::str() {
+    std::stringstream sst;
+    sst << schema;
+    sst << "://";
+    sst << address;
+    if ((schema == "http" and port != 80) or
+        (schema == "https" and port != 443)) {
```

Comment:
@hellais one issue with fsrn.org was that we constructed the URL contained `443` and this lead to a redirect loop; the fix was to make sure no port was added when the port is the default one for a schema.